testing/communication/AppRouteShape.ttl

```diff
@@ -0,0 +1,86 @@
+		sh:prefix "ids" ;
+		sh:namespace "https://w3id.org/idsa/core/"^^xsd:anyURI ;
+	] .
+
+
+shapes:AppRouteShape
+	a sh:NodeShape ;
+	sh:targetClass ids:AppRoute ;
+
+	sh:property [
+		a sh:PropertyShape ;
+		sh:path ids:routeDeployMethod ;
+    sh:datatype xsd:string ;
+		sh:minCount 1 ;
+		sh:severity sh:Violation ;
+		sh:message "<https://raw.githubusercontent.com/International-Data-Spaces-Association/InformationModel/master/testing/communication/AppRouteShape.ttl> (AppRouteShape): An ids:routeDeployMethod property must point from an ids:AppRoute to at least one xsd:string."@en ;
+	] ;
+
+	sh:property [
+		a sh:PropertyShape ;
+		sh:path ids:routeConfiguration ;
+    sh:datatype xsd:string ;
+		sh:severity sh:Violation ;
+		sh:message "<https://raw.githubusercontent.com/International-Data-Spaces-Association/InformationModel/master/testing/communication/AppRouteShape.ttl> (AppRouteShape): An ids:routeConfiguration property must point from an ids:AppRoute an xsd:string."@en ;
+
+	] ;
+
+  sh:property [
+  a sh:PropertyShape ;
+  sh:path ids:routeDescription ;
+  sh:datatype xsd:string ;
+  sh:severity sh:Violation ;
+  sh:message "<https://raw.githubusercontent.com/International-Data-Spaces-Association/InformationModel/master/testing/communication/AppRouteShape.ttl> (AppRouteShape): An ids:routeDescription property must point from an ids:AppRoute an xsd:string."@en ;
+  ] ;
+
+	sh:property [
+		a sh:PropertyShape ;
+		sh:path ids:subRoute ;
+    sh:class ids:EndToEndRoute ;
+		sh:severity sh:Violation ;
+		sh:message "<https://raw.githubusercontent.com/International-Data-Spaces-Association/InformationModel/master/testing/communication/AppRouteShape.ttl> (AppRouteShape): An ids:subRoute property must point from an ids:AppRoute an ids:EndToEndRoute."@en ;
+	] ;
+
+	sh:property [
+		a sh:PropertyShape ;
+		sh:path ids:routeConfiguration ;
+    sh:datatype xsd:string ;
+		sh:severity sh:Violation ;
+		sh:message "<https://raw.githubusercontent.com/International-Data-Spaces-Association/InformationModel/master/testing/communication/AppRouteShape.ttl> (AppRouteShape): An ids:routeConfiguration property must point from an ids:AppRoute an xsd:string."@en ;
+	] ;
+
+	sh:property [
+		a sh:PropertyShape ;
+		sh:path ids:appRouteStart ;
+    sh:class ids:AppEndpoint ;
+		sh:severity sh:Violation ;
+		sh:message "<https://raw.githubusercontent.com/International-Data-Spaces-Association/InformationModel/master/testing/communication/AppRouteShape.ttl> (AppRouteShape): An ids:appRouteStart property must point from an ids:AppRoute an ids:AppEndpoint."@en ;
+	] ;
+
+	sh:property [
+		a sh:PropertyShape ;
+		sh:path ids:appRouteEnd ;
+    sh:class ids:AppEndpoint ;
+		sh:severity sh:Violation ;
+		sh:message "<https://raw.githubusercontent.com/International-Data-Spaces-Association/InformationModel/master/testing/communication/AppRouteShape.ttl> (AppRouteShape): An ids:appRouteEnd property must point from an ids:AppRoute an ids:AppEndpoint."@en ;
+	] ;
+.
```

Comment:
there are still ids:appRouteOutput, ids:appRouteBroker, ids:appRouteClearingHouse and ids:requiredDataApp, why are they not in this shape file?

Reply:
This is simply because right now, I am "only" getting rid of all differences appearing in the output of the new codegen tool and the current nightly version of the java classes. You could say, these changes are my priorty right now.

But of course you are completely right, that they should also be added, even if they don't have any cardinality restrictions. I only wrote the additional shapes that I needed to eliminate differences so far.

Reply:
It might be good for us to come up with two SPARQL queries. One finds shapes which are for non-existing classes or properties. The other one searches for classes and properties which have no shape. Shouldn't be terribly difficult, but help with the review